src/server/array_func.c

```diff
@@ -625,19 +590,20 @@ subst_array_index(job *pjob, char *path)
 {
 	char *pindorg;
 	char cvt[10];
-	char trail[MAXPATHLEN+1];
+	char trail[MAXPATHLEN + 1];
+	job *ppjob = pjob->ji_parentaj;
 	static char index_tag[] = PBS_FILE_ARRAY_INDEX_TAG;
 
-	if (pjob->ji_parentaj == NULL)
+	if (ppjob == NULL)
 		return path;
 	if ((pindorg = strstr(path, index_tag)) == NULL)
-		return path;	/* unchanged */
+		return path; /* unchanged */
 
-	sprintf(cvt, "%d", pjob->ji_parentaj->ji_ajtrk->tkm_tbl[pjob->ji_subjindx].trk_index);
+	sprintf(cvt, "%d", SJ_TBLIDX_2_IDX(ppjob, pjob->ji_subjindx));
```

Comment:
Could you make cvt 21 or 22 characters (or maybe just 100).  20 is the length of a max signed long, 21 is the size of a max unsigned long.